google-datacatalog-kafka-connector/tests/google/datacatalog_connectors/kafka/prepare/datacatalog_entry_factory_test.py

```diff
@@ -22,42 +25,51 @@ def setUp(self):
         self.__entry_factory = DataCatalogEntryFactory(
             self.__PROJECT_ID, self.__LOCATION_ID, self.__METADATA_SERVER_HOST,
             self.__ENTRY_GROUP_ID)
+        self.addTypeEqualityFunc(datacatalog_v1beta1.types.Entry,
+                                 self.compare_entries)
 
     def test_topic_metadata_should_be_converted_to_datacatalog_entries(
             self, entry_path):
         entry_path.return_value = \
             DataCatalogEntryFactoryTestCase.__MOCKED_ENTRY_PATH
 
-        metadata = utils.Utils.convert_json_to_object(self.__MODULE_PATH,
-                                                      'test_metadata.json')
+        metadata = utils.Utils.convert_json_to_object(
+            self.__MODULE_PATH, 'test_metadata_one_topic.json')
         topics = metadata[MetadataDictKeys.TOPICS]
 
         for topic in topics:
             entry_id, entry = self.__entry_factory.make_entry_for_topic(topic)
+            expected_entry = test_utils.get_test_topic_entry()
             self.assertIsNotNone(entry_id)
-            self.assertEqual('kafka_topic', entry.user_specified_type)
-            self.assertEqual('kafka', entry.user_specified_system)
-            self.assertEqual(
-                DataCatalogEntryFactoryTestCase.__MOCKED_ENTRY_PATH,
-                entry.name)
-            self.assertIn(
-                DataCatalogEntryFactoryTestCase.__METADATA_SERVER_HOST,
-                entry.linked_resource)
-            self.assertIn(topic, entry.display_name)
+            self.assertEqual(entry, expected_entry)
 
     def test_cluster_metadata_should_be_converted_to_dc_entries(
             self, entry_path):
         entry_path.return_value = \
             DataCatalogEntryFactoryTestCase.__MOCKED_ENTRY_PATH
         metadata = utils.Utils.convert_json_to_object(self.__MODULE_PATH,
                                                       'test_metadata.json')
-        cluster_id = metadata[MetadataDictKeys.CLUSTER_ID]
         entry_id, entry = self.__entry_factory.make_entry_for_cluster(metadata)
+        expected_entry = test_utils.get_test_cluster_entry()
         self.assertIsNotNone(entry_id)
-        self.assertEqual('kafka_cluster', entry.user_specified_type)
-        self.assertEqual('kafka', entry.user_specified_system)
-        self.assertEqual(DataCatalogEntryFactoryTestCase.__MOCKED_ENTRY_PATH,
-                         entry.name)
-        self.assertIn(DataCatalogEntryFactoryTestCase.__METADATA_SERVER_HOST,
-                      entry.linked_resource)
-        self.assertIn(cluster_id, entry.display_name)
+        self.assertEqual(entry, expected_entry)
+
+    def compare_entries(self, entry1, entry2, msg=None):
+        if not msg:
```

Comment:
I wonder if there is a way to not compare field-by-field by hand. Perhaps comparing string representations of Entries would be enough?

Reply:
Done! Thanks for your suggestion, I didn't know it was an option.